### ISSUE TITLE
Suggest an appropriate definition for "Change" of a lab test result

### ISSUE THREAD

**Issue by raikens1** [HealthRex/CDSS#130]
We want a function (C: RxR -> {0,1}) for change between u (new measurement) and v (last measurement).  Here is a list of properties for C that might be nice
1. Easy to calculate
2. Non-parametric (this may be hard)
3. Sensitive enough to give a good balance of change/not change
4. Invariant to mean (a), variance (b), and shape (c) of distribution
5. Reflect location in p.d.f?  (e.g. If P(u) is small, do we care more about change than if P(u) is large?)

**Comment by raikens1** [comment]
Seems as though defining change involves selecting a “distance” function (need not be a true “metric” in the mathematical sense), and selecting a threshold q.

**Method ideas:**

**Interval** C(u, v) = I(|u-v| < q): 1, 3 if calibrated, 4(a), 4(b) only if scaled 4(c)?
**Percent/ratio** C(u, v) = I(|1-u/v| < q): 1, 3 if calibrated, 4(a), 4(b) only if scaled, (5-ish) change is smaller if v is small, larger if v is large
 - Makes some sense if u and v are exponentially distributed, but not if u and v are normal
**SD Interval** C(u,v) = I(|u-v|/s < q), where s is s.e. over training set: 1, 3 if calibrated, 4(a), 4(b), 4(c)?
 - Equivalent to a threshold of Mahalanobis distance, or Interval under se scaling
**PDF distance** C(u,v) = I(|P(u) - P(v)| < q): 1??, 3 if calibrated, 4(a, b, c), (5) change is greatest when p.d.f is steepest
 - This seems nice, but how to calculate? Need to get an approximation of P from the training data.
**PDF ratio distance** C(u,v) = I(|1 - P(u)/P(v)| < q): 1??, 3 if calibrated, 4(a, b, c), (5) more sensitive to change when P(v) is small
 - Again, how to calculate?
 - Could do inverse ratio: C(u,v) = I(|1-P(v)/P(u)|, which is more sensitive when P(u) is small)
**CDF distance** C(u, v) = I(|F(u) - F(v)| < q): 1, 3 if calibrated, 4(a,b,c), (5) change is greatest when P(u) and P(v) are large
- If either u or v is “normal”, it probably has high probability, which means F is changing rapidly (and we are sensitive to small distances between u and v).  This allows us to pick up on cases where one of u or v is normal and the other is not (nice!) but we also will estimate change to be large when both u and v are normal, and is very insensitive when both u and v are abnormal (possibly not nice). 
Can calculate empirically based on the number of training examples between u and v.

**Comment by raikens1** [comment]
This is a good start.  Going to implement percent, interval, and SD interval for now.
